onnxruntime/core/mlas/lib/platform.cpp

```diff
@@ -425,3 +434,58 @@ Return Value:
     return MLAS_DEFAULT_PREFERRED_BUFFER_ALIGNMENT;
 #endif
 }
+
+#if defined(MLAS_TARGET_ARM64) && defined(__linux__)
+static MlasCoreType* mlas_coretype_tbl = nullptr;
+static uint32_t mlas_coretype_tbl_size = 0;
+static std::once_flag mlas_init_coretype_tbl;
+#endif
+
+MlasCoreType
+MlasGetCoreType()
+{
+
+#if defined(MLAS_TARGET_ARM64) && defined(__linux__)
+    std::call_once(mlas_init_coretype_tbl, []() {
+        mlas_coretype_tbl_size = std::thread::hardware_concurrency();
+        if (mlas_coretype_tbl_size == 0) {
+            return;
+        }
+        mlas_coretype_tbl = (MlasCoreType*)malloc(sizeof(MlasCoreType) * mlas_coretype_tbl_size);
+        for (uint32_t i = 0; i < mlas_coretype_tbl_size; i++) {
+            mlas_coretype_tbl[i] = mlas_core_unknown;
+        }
+    });
+
+    if (mlas_coretype_tbl_size == 0) {
+        // functionality missing, return default
+        return mlas_core_big;
+    }
+
+    unsigned cpu = 0;
+    if (syscall(__NR_getcpu, &cpu, NULL, NULL) != 0) {
+            // failed to detect current core id. give up
+            return mlas_core_big;
+    }
+
+    if (cpu >= mlas_coretype_tbl_size) {
+        return mlas_core_big;
+    }
+
+    auto core_type = mlas_coretype_tbl[cpu];
+    if (core_type == mlas_core_unknown) {
+        auto uarch = MLAS_CPUIDINFO::GetCPUIDInfo().GetCurrentUarch();
+        if (uarch == cpuinfo_uarch_cortex_a53 || uarch == cpuinfo_uarch_cortex_a55r0 ||
+            uarch == cpuinfo_uarch_cortex_a55) {
+            core_type = mlas_core_little;
+        } else {
+            core_type = mlas_core_big;
+        }
+        mlas_coretype_tbl[cpu] = core_type;
+    }
```

Comment:
How about moving this logic and the above std::call_once to MLAS_CPUIDINFO/CPUInfo. In this way, all those  initialization will be done at the construction stage of CPUInfo.